### PR TITLE
ENH: Added a color legend item to the slice view context menu for volumes

### PR DESCRIPTION
**Pull request by MichaelColonel**
The item is visible if slice view has a volume on it, and volume has valid color node and primary display node. Part of the #6111 issue.

Example of the context menu item.

image

### CODE REVIEW

File: Modules/Loadable/Volumes/SubjectHierarchyPlugins/qSlicerSubjectHierarchyVolumesPlugin.cxx [hunk 1]
```diff
@@ -1298,3 +1334,36 @@ void qSlicerSubjectHierarchyVolumesPlugin::setVolumePreset(const QString& preset
...
 }
+
+//---------------------------------------------------------------------------
+void qSlicerSubjectHierarchyVolumesPlugin::toggleVisibilityForCurrentItem(bool on)
+{
+  Q_D(qSlicerSubjectHierarchyVolumesPlugin);
+  if (d->SelectedVolumeNode)
+    {
+    // Show color legend display node
+    vtkMRMLDisplayNode* displayNode = d->SelectedVolumeNode->GetDisplayNode();
```
Comment (lassoan) [inline Modules/Loadable/Volumes/SubjectHierarchyPlugins/qSlicerSubjectHierarchyVolumesPlugin.cxx:1345]: You need to call `GetVolumeDisplayNode()` to make sure you don't get a volume rendering or color legend display node.

```suggestion
    vtkMRMLDisplayNode* displayNode = d->SelectedVolumeNode->GetVolumeDisplayNode();
```

File: Modules/Loadable/Volumes/SubjectHierarchyPlugins/qSlicerSubjectHierarchyVolumesPlugin.cxx [hunk 2]
```diff
@@ -145,6 +149,15 @@ void qSlicerSubjectHierarchyVolumesPluginPrivate::init()
   this->ResetViewOrientationOnShowAction->setCheckable(true);
   this->ResetViewOrientationOnShowAction->setChecked(false);
 
+  // Add color legend action
+
+  this->ShowColorLegendAction = new QAction("Show color legend", q);
```
Comment (lassoan) [inline Modules/Loadable/Volumes/SubjectHierarchyPlugins/qSlicerSubjectHierarchyVolumesPlugin.cxx:154]: Mark the string translatable by using `tr`.

```suggestion
  this->ShowColorLegendAction = new QAction(tr("Show color legend"), q);
```

**Review by lassoan — COMMENTED** [review]
It looks good, I've just added a few nitpicks. Please also add a line to the documentation where the options for show/hide color legend is listed (https://slicer.readthedocs.io/en/latest/user_guide/modules/colors.html#show-color-legend).

File: Modules/Loadable/Volumes/SubjectHierarchyPlugins/qSlicerSubjectHierarchyVolumesPlugin.cxx (hunk 2 above)
Comment (MichaelColonel) [reply Modules/Loadable/Volumes/SubjectHierarchyPlugins/qSlicerSubjectHierarchyVolumesPlugin.cxx]: > It looks good, I've just added a few nitpicks. Please also add a line to the documentation where the options for show/hide color legend is listed (https://slicer.readthedocs.io/en/latest/user_guide/modules/colors.html#show-color-legend).

Description added.

**Review by lassoan — APPROVED** [review]
Thank you, it is great. I'll merge it now.

**Pull request merged**